go/libraries/doltcore/sqle/expreval/expression_evaluator.go

```diff
@@ -173,16 +173,16 @@ func GetComparisonType(be expression.BinaryExpression) ([]*expression.GetField,
 	}
 
 	var compType ComparisonType
-	if len(consts) == 2 {
-		compType = ConstConstCompare
-	} else if len(variables) == 2 {
+	if len(variables) == 2 {
 		compType = VariableVariableCompare
 	} else if len(variables) == 1 {
 		if len(consts) == 1 {
 			compType = VariableConstCompare
 		} else if len(consts) > 1 {
 			compType = VariableInLiteralList
 		}
+	} else if len(consts) == 2 {
+		compType = ConstConstCompare
```

Comment:
This is a bug fix. if an "IN" clause has exactly two values then this was broken.